tests/unit/components/sl-modal-test.js

```diff
@@ -0,0 +1,339 @@
+
+    Ember.run( () => {
+        component.show();
+    });
+
+    this.$( '.close' ).click();
+});
+
+test( 'Modal registered on modal service', function( assert ) {
+    let registerSpy = sinon.spy();
+
+    let mockModalService = {
+        register: registerSpy,
+        find: function(){ }
+    };
+
+    component = this.subject({
+        name: 'demo',
+        modalService: mockModalService
+    });
+
+    this.render();
+
+    assert.ok(
+        registerSpy.calledOnce,
+        'Register called on modal service'
+    );
+});
+
+test( 'Backdrop is hidden when backdrop property is set to false', function ( assert ) {
+    let openDone = assert.async();
+
+    assert.expect( 1 );
+
+    component = this.subject({
+        afterShow: 'modalOpen',
+        backdrop: false,
+        modalService: mockModalService,
+        targetObject: {
+            modalOpen() {
+                assert.equal(
+                    $( '.modal-backdrop' ).length,
+                    0
+                );
+                openDone();
+            }
+        }
+    });
+
+    this.render();
+
+    Ember.run( () => {
+        component.show();
+    });
+});
+
+test( 'Backdrop is shown by default', function( assert ) {
+    component = this.subject({
+        modalService: mockModalService
+    });
+
+    this.render();
+
+    Ember.run( () => {
+        component.show();
+    });
+
+    assert.equal(
+        $( '.modal-backdrop' ).length,
+        1
+    );
+});
+
+test( 'Modal is unregistered after destroy', function( assert ) {
+    let spyUnregister = sinon.spy();
+
+    component = this.subject({
+        modalService: mockModalService,
+        name: 'testDestroy',
+        unregister: spyUnregister
+    });
+
+    this.render();
+
+    Ember.run( () => {
+        component.destroy();
+        assert.ok( spyUnregister.calledOnce );
+    });
+});
+
+test( 'Fade class is present when animated is set to true', function( assert ) {
+    component = this.subject({
+        modalService: mockModalService
+    });
+
+    assert.ok(
+        this.$().hasClass( 'fade' )
+    );
+});
+
+test( 'Fade class is absent when animated is set to false', function( assert ) {
+    component = this.subject({
+        modalService: mockModalService,
+        animated: false
+    });
+
+    assert.ok(
+        !this.$().hasClass( 'fade' )
+    );
+
+});
+
+test( 'aria-describedby attribute binding', function( assert ) {
+    let describedBy = 'targetId';
+
+    component = this.subject({
+        modalService: mockModalService,
+        'aria-describedby': describedBy
+    });
+
+    assert.equal(
+        this.$().attr( 'aria-describedby' ),
+        describedBy
+    );
+});
+
+test( 'aria-hidden is true', function( assert ) {
+   component = this.subject({
+       modalService: mockModalService
+   });
+
+   this.render();
+
+   assert.equal(
+       this.$().attr( 'aria-hidden' ),
+       'true'
+   );
+});
+
+test( 'aria-labelledby is set', function( assert ) {
```

Comment:
Actually now that I think about it, this makes sense here. This attribute is pulled off the `sl-modal-header`. I will however add tests for `aria-labelledby` on the `sl-modal-header` component.